pkgs/build-support/rust/fetchcargo.nix

```diff
@@ -23,9 +23,11 @@ stdenv.mkDerivation {
 
     ${cargoUpdateHook}
 
-    cargo vendor
+    mkdir $out
+    cargo vendor > $out/config
```

Comment:
this file contains the full path to `./vendor` iirc which is a build impurity. So I think another sed like line 66 of pkgs/build-support/rust/default.nix is necessary to ensure this is really a fixed output derivation.
Also (nitpick) you can use the syntax `cargo vendor $out/vendor > $out/config` to spare a copy.